training/intel-bootc/Containerfile

```diff
@@ -1,95 +1,163 @@
+    make -j$(nproc) && make install && cd / && rm -rf /tmp/libfabric-${LIBFABRIC_VERSION}.tar.bz2 /tmp/libfabric-${LIBFABRIC_VERSION}
+#Hccl wrapper
+RUN curl -L -o /tmp/main.zip https://github.com/HabanaAI/hccl_ofi_wrapper/archive/refs/heads/main.zip && \
+    unzip /tmp/main.zip -d /tmp && \
+    cd /tmp/hccl_ofi_wrapper-main && \
+    make && cp -f libhccl_ofi_wrapper.so /usr/lib/habanalabs/libhccl_ofi_wrapper.so && \
+    cd / && \
+    rm -rf /tmp/main.zip /tmp/hccl_ofi_wrapper-main
 
-ARG VENDOR=''
-LABEL vendor=${VENDOR}
-LABEL org.opencontainers.image.vendor=${VENDOR}
+FROM ${BASEIMAGE}
+ARG DRIVER_VERSION="1.17.1-40"
+ARG ARTIFACTORY_URL="vault.habana.ai"
+ARG REPO_TOKEN
 
+USER root
 COPY --from=builder /home/builder/usr/src/habanalabs-${DRIVER_VERSION}/drivers/accel/habanalabs/habanalabs.ko /tmp/extra/habanalabs.ko
 COPY --from=builder /home/builder/usr/src/habanalabs-${DRIVER_VERSION}/drivers/infiniband/hw/hbl/habanalabs_ib.ko /tmp/extra/habanalabs_ib.ko
 COPY --from=builder /home/builder/usr/src/habanalabs-${DRIVER_VERSION}/drivers/net/ethernet/intel/hbl_cn/habanalabs_cn.ko /tmp/extra/habanalabs_cn.ko
 COPY --from=builder /home/builder/usr/src/habanalabs-${DRIVER_VERSION}/drivers/net/ethernet/intel/hbl_en/habanalabs_en.ko /tmp/extra/habanalabs_en.ko
-COPY --from=builder /home/builder/lib/firmware/habanalabs /tmp/firmware/habanalabs
-COPY --from=builder /home/builder/usr/bin/hl-smi /usr/bin/hl-smi
-
+COPY --from=builder /home/builder/etc/ /etc/
+COPY --from=builder /home/builder/lib/firmware/habanalabs/gaudi/ /lib/firmware/habanalabs/gaudi/
+COPY --from=builder /home/builder/usr/sbin /usr/sbin/
+COPY --from=libbuilder /usr/lib/habanalabs/libhccl_ofi_wrapper.so /usr/lib/habanalabs/libhccl_ofi_wrapper.so
+COPY --from=libbuilder /opt/habanalabs/libfabric-1.20.0 /opt/habanalabs/libfabric-1.20.0
+COPY scripts/os_dependencies.sh /tmp/os_dependencies.sh
+
+#Install python3.11 and other build stuff
+RUN /tmp/os_dependencies.sh \
+    && dnf install -y \
+    python3.11 \
+    python3.11-pip \
+    python3.11-devel \
+    git \
+    make \
+    gcc-c++ \
+    unzip \
+    && dnf clean all && rm -rf /var/cache/yum 
 
-COPY duplicated/common/usr /usr
+#Build ninja-build
+RUN git clone https://github.com/ninja-build/ninja.git \
+    && cd ninja \
+    && ./configure.py --bootstrap
 
 RUN . /etc/os-release \
     && export OS_VERSION_MAJOR=$(echo ${VERSION} | cut -d'.' -f 1) \
     && export KERNEL_VERSION=$(rpm -q --qf '%{VERSION}-%{RELEASE}' kernel-core) \
     && export TARGET_ARCH=$(rpm -q --qf '%{ARCH}' kernel-core) \
-    && mv /tmp/extra /lib/modules/${KERNEL_VERSION}.${TARGET_ARCH} \
-    && mv /tmp/firmware/habanalabs /lib/firmware \
-    && depmod -a ${KERNEL_VERSION}.${TARGET_ARCH}
-
-RUN mv /etc/selinux /etc/selinux.tmp \
-    dnf install -y ${EXTRA_RPM_PACKAGES} \
+    && dnf -y update --exclude=kernel* --exclude=microcode_ctl \
+    && dnf install -y --nodocs --allowerasing --best \
+    libogg-devel \
```

Comment:
Are these devel rpms required in production or only during container build?

Reply:
Just for building. Will remove them.